tests/test_export.py

```diff
@@ -57,3 +61,155 @@ def test_export_datasets(archive, layer, geometry, data_archive, cli_runner):
                 continue
 
             assert field_name[0:10] in output
+
+
+def test_dataset_creation_options(data_archive, cli_runner):
```

Comment:
> Should add some more test cases:
using dataset creation options
using layer creation options
...

That's all the suggested tests I've got to so far - there are more suggested tests in the comments on the last PR, and I'll get to them in the next PR.
But, I also added tests for the --primary-key-as-field and --primary-key-as-fid options, which I hope helps document them.